clusterloader2/state/tuningsets.go

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package state
+
+import (
+	"fmt"
+	"sync"
+
+	"k8s.io/perf-tests/clusterloader2/api"
+)
+
+// tuningSets represents set of available tuning sets.
+type tuningSets struct {
+	tuningSets map[string]*api.TuningSet
+	lock       sync.RWMutex
+}
+
+func (ts *tuningSets) getTuningSet(name string) (*api.TuningSet, error) {
```

Comment:
About naming: "tuningSets.GetTuningSet" sounds strange.

It probably should be "tuningSets.get()" instead.

Same for set and in other relevant places in this whole PR.